irispy/spectrograph.py

```diff
@@ -229,17 +229,29 @@ def __init__(self, filenames, spectral_windows="All"):
 
     def __repr__(self):
         spectral_window = self.spectral_windows["name"][0]
-        spectral_windows_info = "".join(
-            ["\n    {0}\n        (raster axis: {1}, slit axis: {2}, spectral axis: {3})".format(
-            name, len(self.data[name].raster_axis), len(self.data[name].slit_axis),
-            len(self.data[name].spectral_axis)) for name in self.spectral_windows["name"]])
+        spectral_windows_info = ""
+        for name in self.spectral_windows['name']:
+            shape = self.data[name].shape
+            spectral_windows_info = spectral_windows_info + ")\n    {0}\n           (".format(name)
+            for i, dim in enumerate(self.data[name].dims):
+                spectral_windows_info = spectral_windows_info + \
+                        "{0}".format(dim) + ": {0}  ".format(shape[i])
+
+        instance_period_data = self.data[spectral_window].time.values
+        if type(instance_period_data) is not np.datetime64:
+            instance_period_start = instance_period_data[0]
+            instance_period_end   = instance_period_data[-1]
+        else:
+            instance_period_start = instance_period_data
+            instance_period_end   = instance_period_data
+
         return "<iris.IRISRaster instance\nOBS ID: {0}\n".format(self.meta["observation ID"]) + \
                "OBS Description: {0}\n".format(self.meta["observation description"]) + \
                "OBS period: {0} -- {1}\n".format(self.meta["observation start"], self.meta["observation end"]) + \
-               "Instance period: {0} -- {1}\n".format(self.data[spectral_window].time.values[0],
```

Comment:
I think as above, the result will be an array of length 1.  So giving the using the 0th index will work.  If this is not the case, it should be.  We don't want the type of an attribute to change just because the length changes.  But again, it would good if you could confirm this.